### PR TITLE
allow variables in numeric rules to be valid

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

**Comment by ericnewcomer** [comment]
unit test?

File: static/coffee/flows/validation.coffee [hunk 1]
```diff
@@ -5,6 +5,7 @@ app = angular.module('temba.validation', [])
 #============================================================================
 REGEX_NUMBER = /^\-?\d+[\.\d+]*$/
 REGEX_ALPHANUM = /^[a-z\d\-_\s]+$/i
+REGEX_VARIABLE = /^[ ]*@.+$/i
```
Comment (norkans7) [inline static/coffee/flows/validation.coffee]: g modifier on regex, RegExp.test will keep going through the string until the end trying to find the next occurence, so not including it here is the right thing

Comment (nicpottier) [inline static/coffee/flows/validation.coffee]: I still don't like this.. the explanation doesn't really make sense. /g shouldn't make a difference here, which to me points to something not understood and a possible bug.

standalone I can create a regex using /g and without on a pattern and they behave the same with ```test```

Comment (norkans7) [inline static/coffee/flows/validation.coffee]: after testing in the console, I think using g will break sometime unless we add loop until we make sure test has reached the end of the string each time.


re-g-modifier

Comment (nicpottier) [inline static/coffee/flows/validation.coffee]: Aha, now we are getting somewhere. Turns out the /g flag modifies the regex itself (it has a side effect) as you found:
   http://stackoverflow.com/questions/604860/interesting-test-of-javascript-regexp

So ya, removing /g/ is the right thing. :)

Comment (nicpottier) [inline static/coffee/flows/validation.coffee]: Best description of the behavior here:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/lastIndex

**Comment by ericnewcomer** [comment]
It'd be good to add a failing validation case to your test as well.

**Comment by norkans7** [comment]
Okay

**Comment by nicpottier** [comment]
Excellent.

**Pull request merged**